eda_analysis/eda_analysis.py

```diff
@@ -46,11 +49,43 @@ def describe_na_values(dataframe):
     numpy.ndarray
         A 2d Numpy Array of 1's and 0's, corresponding to the value of each entry in the dataframe.
         0 represents an NA value, 1 represents a non-NA value.
+    Examples:
+    ---------
+    >>> no_na_dataframe = pd.DataFrame({
+                                    "col_1": [0, 2],
+                                    "col_2": [0.5, 0.1],
+                                    "col_3": ["a", "b"]
+                                    })
+    >>> describe_na_variable(no_na_dataframe)
+    numpy.ndarray([1, 1],
+                  [1, 1],
+                  [1, 1])
+
+    >>>  na_numerical_dataframe = pd.DataFrame({
+                                           "col_1": [0, 2],
+                                           "col_2": [numpy.nan, 0.1],
+                                           "col_3": ["a", "b"]
+                                           }),
+    >>> describe_na_variable(na_numerical_dataframe)
+    numpy.ndarray([1, 1],
+                  [0, 1],
+                  [1, 1])
+    >>>  na_categorical_dataframe = pd.DataFrame({
+                                           "col_1": [0, 2],
+                                           "col_2": [0.5, 0.1],
+                                           "col_3": [np.nan, "b"]
+                                           }),
+    >>> describe_na_variable(na_numerical_dataframe)
+    numpy.ndarray([1, 1],
+                  [1, 1],
+                  [0, 1])
+
     '''
```

Comment:
Include some defensive  coding like these :
```
    if not isinstance(dataframe, pd.DataFrame):
        raise Exception("The value of the argument 'dataframe' must be of type 'pandas.DataFrame'")
    
    if not isinstance(cat_vars,list) or not all(isinstance(x, str) for x in cat_vars):
        raise Exception("The value of the argument 'cat_vars' must be a list of strings")
```
Defensive coding is one of the requirement and even your test case should also have some negative testcase for these